### ISSUE TITLE
Mock documentation requirements

### ISSUE THREAD

**Issue by gianscarpe** [kornia/kornia#750]
## 🚀 Feature

Mock requirments for building documentation

## Motivation
I think it would be cleaner to mock the requirments for the documentation. Documentation building shouldn't require `opencv` and similar.

## Pitch
We can read the `requirements.txt` file and add the packages to `conf.py` under  `autodoc_mock_imports`

## Additional context
I implemented something similar in [one of my projects](https://github.com/gianscarpe/event-camera/blob/master/docs/source/conf.py)

**Comment by edgarriba** [comment]
@gianscarpe the reason for having e.g `opencv` in the docs is because we build some tutorials inside the docs. This is gonna change since we'll move all the tutorials in a separated repo called tutorials which will generate its own docs. I'll provide a hello world I hope today. Then we could remove any heavy dependency from docs itself.

**Comment by edgarriba** [comment]
@shijianjian @Borda do you think we can adopt this ? and what benefits can give to use ?

**Comment by Borda** [comment]
> do you think we can adopt this ? and what benefits can give to use ?

yes, you can the advantage is that whenever you build docs on a new env (like ReadTheDocs) no dependencies are needed to be installed...
